### PR TITLE
#4422 Fix the main class of shabondi does not exist

### PR DESCRIPTION
**Pull request by vitojeng**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #4422)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-agent/src/main/scala/oharastream/ohara/agent/ShabondiCollie.scala [hunk 1]
```diff
@@ -103,7 +103,7 @@ trait ShabondiCollie extends Collie {
                     }
                     k + "=" + value
                 }.toSeq
-                doCreator(executionContext, containerInfo, newNode, routes, arguments)
+                doCreator(executionContext, containerInfo, newNode, routes, Seq(creation.shabondiClass) ++ arguments)
```
Comment (chia7712) [inline ohara-agent/src/main/scala/oharastream/ohara/agent/ShabondiCollie.scala:106]: great

File: ohara-it/src/test/scala/oharastream/ohara/it/shabondi/TestShabondi.scala [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+  private[this] var topicApi: TopicApi.Access         = _
+  private[this] var jarApi: FileInfoApi.Access        = _
+
+  private[this] var shabondiApi: ShabondiApi.Access = _
+  private[this] var bkKey: ObjectKey                = _
+  private[this] var brokerConnProps: String         = _
+
+  @Before
+  def setup(): Unit = {
+    zkApi = ZookeeperApi.access.hostname(configuratorHostname).port(configuratorPort)
```
Comment (chia7712) [inline ohara-it/src/test/scala/oharastream/ohara/it/shabondi/TestShabondi.scala:54]: 可以的話盡量讓參數宣告成val，也就是這些可以放在建構式內就先初始化完成

Comment (vitojeng) [reply ohara-it/src/test/scala/oharastream/ohara/it/shabondi/TestShabondi.scala]: done

File: ohara-it/src/test/scala/oharastream/ohara/it/shabondi/TestShabondi.scala [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+  private[this] var containerApi: ContainerApi.Access = _
+  private[this] var topicApi: TopicApi.Access         = _
+
+  private[this] var shabondiApi: ShabondiApi.Access = _
+  private[this] var bkKey: ObjectKey                = _
+
+  @Before
+  def setup(): Unit = {
+    val zkApi = ZookeeperApi.access.hostname(configuratorHostname).port(configuratorPort)
+    bkApi = BrokerApi.access.hostname(configuratorHostname).port(configuratorPort)
```
Comment (chia7712) [inline ohara-it/src/test/scala/oharastream/ohara/it/shabondi/TestShabondi.scala:45]: 我應該沒表達清楚，我先前的意思是說這支class裡面大多數的"var"都可以改成"val"，也就是他們可以在constructor階段就初始化完成，而不用等到before這個階段

Comment (vitojeng) [reply ohara-it/src/test/scala/oharastream/ohara/it/shabondi/TestShabondi.scala]: Got it, thanks for the reminder.

**Comment by vitojeng** [comment]
not related error

**Pull request merged**